extensions/warp-fs-memory/src/native.rs

```diff
@@ -70,15 +70,13 @@ impl Constellation for MemorySystem {
 
         let name = name.to_string();
 
-        let stream = ConstellationProgressStream(
-            futures::stream::once(async move {
-                Progression::ProgressComplete {
-                    name,
-                    total: Some(bytes as _),
-                }
-            })
-            .boxed(),
-        );
+        let stream = futures::stream::once(async move {
```

Comment:
if the stream always returns only 1 item, I'd not return a stream but a future. The client code can easily convert a stream to a future if it needs to.

Reply:
I do not follow. Although `warp-fs-memory` is due for a rewrite (its left there mostly for testing when the components were more interchangeable and used for testing purpose), returning a future instead of a stream would break the overall api. Using a stream, even if it is only one item in the stream, would keep it compatible with the api.